include/deal.II/matrix_free/face_setup_internal.h

```diff
@@ -1106,11 +1106,11 @@ namespace internal
         {
           const Table<2, unsigned int> orientation =
             ShapeInfo<double>::compute_orientation_table(2);
-          const std::array<unsigned int, 8> inverted_orientations{
-            {0, 1, 6, 3, 4, 5, 2, 7}};
           info.subface_index =
-            orientation[inverted_orientations[exterior_face_orientation]]
-                       [info.subface_index];
+            orientation(ReferenceCells::get_hypercube<dim - 1>()
+                          .get_inverse_combined_orientation(
+                            exterior_face_orientation),
+                        info.subface_index);
```

Comment:
I could not quite convince myself that you can't get here with simplex cells. The code is clearly only correct if you have a hypercube cell. Would you mind adding an assertion?

Reply:
I didn't look that carefully at this when I wrote it but we have enough information to get the correct reference cell in this context - we can do things right and we don't need the assertion. Fixed!